lnwire/update_add_htlc.go

```diff
@@ -152,16 +174,51 @@ func (c *UpdateAddHTLC) Encode(w *bytes.Buffer, pver uint32) error {
 		return err
 	}
 
-	// Only include blinding point in extra data if present.
+	// Construct a slice of all the records that we should include in the
+	// extra data field.
 	var records []tlv.RecordProducer
 
+	// Only include blinding point in extra data if present.
 	c.BlindingPoint.WhenSome(func(b tlv.RecordT[BlindingPointTlvType,
 		*btcec.PublicKey]) {
 
 		records = append(records, &b)
 	})
 
-	err := EncodeMessageExtraData(&c.ExtraData, records...)
+	// Extend the 'records' slice with TLV records from the custom records
+	// field.
+	//
+	// As a sanity check, we will first ensure that the existing records
+	// TLV types are less than the minimum custom records TLV type.
+	for _, recordProducer := range records {
+		record := recordProducer.Record()
+		recordTlvType := uint64(record.Type())
+
+		if recordTlvType >= MinCustomRecordsTlvType {
+			return fmt.Errorf("records contain TLV type that is "+
+				"unexpectedly greater than or equal to the "+
+				"minimum TLV type of custom records: %d",
+				recordTlvType)
+		}
+	}
```

Comment:
What if we ever want to encode a "known" custom type here? Then this validation would disallow it. I think what we want to check instead is just that we don't get any collisions between the known records and the custom ones added by the interceptor.

Reply:
I've added a new method called `CustomRecords.ExtendRecordProducers` which uses the validation that you suggest. It simply checks for no collisions. The `records` slice can contain high TLV types even before extending with records from `CustomRecords`.